### ISSUE TITLE
Why does only the callback return response?

### ISSUE THREAD

**Issue by socketwiz** [t3chnoboy/amazon-product-api#73]
It just took me two hours to figure out that the promise only returns the array of products where as the callback returns both the array of products _and_ the response.  Why is that, and would you accept a pull request to fix the promise?

Callback: https://github.com/t3chnoboy/amazon-product-api/blob/master/lib/index.js#L19
Promise: https://github.com/t3chnoboy/amazon-product-api/blob/master/lib/index.js#L21

In other words, if you use the promise, you can only ever get back the first page of information because it doesn't return the Items object that contains things like:

```xml
<TotalResults>376</TotalResults>
<TotalPages>38</TotalPages>
```

among other things...

**Comment by t3chnoboy** [comment]
Yeah, sorry for the inconsistency 😕 
PRs are welcome! 😄  Or I can take a look on the weekend.

**Comment by masterT** [comment]
Can't Promise return more then one value? How will you do that? Return an array of values?

**Comment by masterT** [comment]
Same problem then issue #49.

**Comment by budda** [comment]
Is this any closer to being fixed and released to npm?
